File: android/common/compose/src/main/kotlin/com/gchristov/newsfeed/android/common/compose/elements/AppText.kt
package com.gchristov.newsfeed.android.common.compose.elements

import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.AnnotatedString
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import com.gchristov.newsfeed.android.common.compose.theme.Theme

@Composable
fun AppText(
    text: String,
    modifier: Modifier = Modifier,
    color: Color = Theme.contentColors.primary,
    textAlign: TextAlign? = null,
    style: TextStyle = Theme.typography.body,
    maxLines: Int = Int.MAX_VALUE,
    overflow: TextOverflow = TextOverflow.Clip,
) {
    androidx.compose.material.Text(
        text = AnnotatedString(text),
        modifier = modifier,
        color = color,
        textAlign = textAlign,
        style = style,
        maxLines = maxLines,
        overflow = overflow
    )
}

File: ios/CommonSwiftUi/Sources/Elements/AppText.swift
import SwiftUI

public struct AppText: View {
    @EnvironmentObject var theme: Theme
    private let text: String
    private let color: Color?
    private let font: Font?
    private let lineLimit: Int?
    private let textAlignment: TextAlignment
    
    public init(
        text: String,
        color: Color? = nil,
        font: Font? = nil,
        lineLimit: Int? = nil,
        textAlignment: TextAlignment = .leading
    ) {
        self.text = text
        self.color = color
        self.font = font
        self.lineLimit = lineLimit
        self.textAlignment = textAlignment
    }
    
    public var body: some View {
        Text(text)
            .font(font ?? theme.typography.body)
            .foregroundColor(color ?? theme.contentColors.primary)
            .lineLimit(lineLimit)
            .multilineTextAlignment(textAlignment)
    }
}
